src/test/java/org/codehaus/stax/test/stream/TestEntityRead.java

```diff
@@ -73,6 +77,52 @@ public void testValidCharEntities()
         assertTokenType(END_ELEMENT, type);
         sr.close();
     }
+    
+    /**
+     * This unit test checks that resolving of surrogate pairs works
+     * as expected, including different ways of writing entities
+     */
+    public void testValidSurrogatePairEntities()
```

Comment:
Shouldn't there be a test or two for invalid ones too?

Reply:
Test for invalid ones is already there - `testInvalidSurrogatePairEntities` method: https://github.com/FasterXML/woodstox/pull/174/files/eb5189d107d911709774c79e1ed445865567d7ff#diff-50c2abc037511dda94147cf808bde7666e630ad76d08151e92ff86fb56b6f17aR466

Reply:
Right, I meant the case when new option was disabled, as per your other comment.